rust/automerge-c/src/item.rs

```diff
@@ -1234,34 +1299,36 @@ pub enum AMvalType {
     ChangeHash = 1 << 5,
     /// A CRDT counter value.
     Counter = 1 << 6,
+    /// A cursor value.
+    Cursor = 1 << 7,
     /// The default tag, not a type signifier.
     Default = 0,
     /// A document value.
-    Doc = 1 << 7,
+    Doc = 1 << 8,
     /// A 64-bit float value.
-    F64 = 1 << 8,
+    F64 = 1 << 9,
     /// A 64-bit signed integer value.
-    Int = 1 << 9,
+    Int = 1 << 10,
     /// A mark.
-    Mark = 1 << 10,
+    Mark = 1 << 11,
     /// A null value.
-    Null = 1 << 11,
+    Null = 1 << 12,
     /// An object type value.
-    ObjType = 1 << 12,
+    ObjType = 1 << 13,
     /// A UTF-8 string view value.
-    Str = 1 << 13,
+    Str = 1 << 14,
     /// A synchronization have value.
-    SyncHave = 1 << 14,
+    SyncHave = 1 << 15,
     /// A synchronization message value.
-    SyncMessage = 1 << 15,
+    SyncMessage = 1 << 16,
     /// A synchronization state value.
-    SyncState = 1 << 16,
+    SyncState = 1 << 17,
     /// A *nix timestamp (milliseconds) value.
-    Timestamp = 1 << 17,
+    Timestamp = 1 << 18,
     /// A 64-bit unsigned integer value.
-    Uint = 1 << 18,
+    Uint = 1 << 19,
     /// An unknown type of value.
-    Unknown = 1 << 19,
+    Unknown = 1 << 20,
```

Comment:
Perhaps I am misunderstanding how these values are used, but redefining these flags seems like it has the potential to break any serialized/binary representations of automerge objects created via the C API. Am I misunderstanding the function of this type?

Reply:
> redefining these flags seems like it has the potential to break any serialized/binary representations of automerge objects created via the C API.

@whereswaldon, those values are not serialized because they're just a way to refer to Rust enum variants from within C; all serialization/deserialization is handled by the underlying Rust library.
I've inserted tags in alphabetical order into that enum before and doing so has never broken a unit test.
Being that everyone is taught to prefer symbolic values over magic numbers I doubt that anyone would choose to use an enum tag's opaque integer value instead of its descriptive name.